CentralComputing/Pod.cpp

```diff
@@ -1,33 +1,61 @@
 #include "Sensor_Package.h"
+#include "Pod.h"
 
 #include <thread>
+#include <csignal>
 
 using namespace std;
 
 Sensor_Package * sensors;
+bool running = true;
+
 /**
 * Gets new sensor values from the XMEGA
 */
 void sensor_loop() {
 	//TODO construct XMEGA transfer object
-	//sensors->update();
+	Xmega_Transfer transfer;
+	while(running) {
+		sensors->update(transfer);
+	}
+
 }
 
+void int_handler(int signum) {
+	running = false;
+}
+
+
 int pod(int argc, char** argv) {
 	
+	signal(SIGINT, int_handler);
 	vector<Sensor_Configuration> configs = vector<Sensor_Configuration>();
 
 	//sets up thermocouple configuration
 	Sensor_Configuration thermo;
 	thermo.type = THERMOCOUPLE;
-	thermo.simulation = 0;
-	thermo.count = 4;
+	thermo.simulation = 1;
+	configs.push_back(thermo);
+
+/*
+	Sensor_Configuration accel;
+	accel.type = ACCELEROMETER;
+	accel.simulation = 1;
+	configs.push_back(accel);
+	*/
 
 	//TODO set up other configurations
+
 	
-	sensors = new Sensor_Package(configs);
+	sensors = new Sensor_Package(configs, false);
```

Comment:
Does it make sense to just have the `Sensor_Package` variable defined in the thread where it will be used? Does it need to be a global variable. Could it just be passed into the thread as an argument? I am generally against global variables when not necessary